lib/Config/Grammar/Dynamic.pm

```diff
@@ -423,6 +423,9 @@ sub parse($$)
 {
     my $self = shift;
     my $file = shift;
+    my $args        = shift;
+
+    $self->{encoding} = $args->{encoding};
```

Comment:
```suggestion
    $self->{encoding} = $args->{encoding} if ref $args eq 'HASH';
```

Reply:
`$self->{encoding}` will be undefined, Which is checked for where it is used.
```
sub _parse_file($$)
{
...

    my $mode = "<";
    $mode .= ":encoding($self->{encoding})" if $self->{encoding};
```
And wouldn't this be enough in case one would want to check:
```
$self->{encoding} = $args->{encoding} if exists $args->{encoding};
```

Reply:
If the check was considered necessary, the same would have to be done in `Grammar.pm` as well.

Reply:
well, try dereferencing $args when it is undefined ... it will not end well

Reply:
Ok, I added the test. For the record: undefined and scalar both worked, but other things like an array ref would indeed fail.

Reply:
wow

```
perl -w -e 'my $x = 1; $x->{hello} = 4; print "$x;$x->{hello}";'
1;4
```
@zaucker can you explain ?

Reply:
so the magic lies in NOT using strict :) 
```
perl -e 'use strict;my $x = 1; $x->{hello} = 4; print "$x;$x->{hello}";'
```